File: producer.py
import pika

# 1. Connect to RabbitMQ
connection = pika.BlockingConnection(
    pika.ConnectionParameters(host='localhost')
)

# 2. Open a channel
channel = connection.channel()

# 3. Send a message
channel.basic_publish(
    exchange='',               # default exchange
    routing_key='demo-pc',  # queue name
    body='Hello from Producer'
)

print("✅ Message sent")

# 4. Close connection
connection.close()

File: Producer.cs
using System;
using System.Text;
using RabbitMQ.Client;

class Program
{
    static void Main(string[] args)
    {
        // 1. Connect to RabbitMQ
        var factory = new ConnectionFactory()
        {
            HostName = "localhost"
        };

        using var connection = factory.CreateConnection();
        using var channel = connection.CreateModel();

        // 2. Message to send
        string message = "Hello from Producer";
        var body = Encoding.UTF8.GetBytes(message);

        // 3. Send message
        channel.BasicPublish(
            exchange: "",
            routingKey: "demo-pc",
            basicProperties: null,
            body: body
        );

        Console.WriteLine("✅ Message sent");
    }
}
